pkg/network/httpx/listener.go

```diff
@@ -0,0 +1,79 @@
+package httpx
+
+import (
+	"errors"
+	"net"
+	"os"
+	"runtime"
+	"strconv"
+	"syscall"
+)
+
+const listenAttempts = 42
+
+type Listener struct {
+	net.Listener
+}
+
+func NewListener(address string, withNextFreePort bool) (*Listener, error) {
```

Comment:
Http/s servers use listeners now just for purpose of getting an actual address (listener port).